integration/integration_suite_test.go

```diff
@@ -128,9 +130,18 @@ var _ = SynchronizedBeforeSuite(func() []byte {
 			false,
 		)
 
-		s3Service = s3.New(session.New(awsConfig), awsConfig)
+		additionalAwsConfig := aws.Config{}
+		if len(awsRoleARN) != 0 {
+			stsConfig := awsConfig.Copy()
+			stsConfig.Endpoint = nil
```

Comment:
Just curious if you could explain to me a bit behind the reasoning for copying the AWS config and nulling out the endpoint in order to construct a session to use with the AWS role arn? Sorry I'm not familiar with AWS but I just want to understand what the code is doing 😅  I think what confuses me here is that we write in the release notes that the AWS role ARN uses the secret and id key but here it looks like we use the entire config? Thanks!

Reply:
My main reason was that the awsConfig seemed to be mainly consisting of useful paramters (see NewAwsConfig in s3client.go). It basically defines some additional thinks like `disableSSL` and `skipSSLVerification`. In `fetchCredentialsForRoleIfDefined` I therefore copy it and remove the endpoint (so that I don't have to create it by copying everything else).
The reason for that is, that the endpoint that's defined here is the s3-endpoint, using this for sts will lead to some confusing errors. I didn't see a good reason the specify it so I did not add a configuration option for it.

But I'm open for suggestions regarding that. It was just a gut-feeling of mine that all the other parameters are helpful here.

Reply:
I see, thanks for explaining!